core/storage/trie/impl/topper_trie_batch_impl.hpp

```diff
@@ -58,8 +59,42 @@ namespace kagome::storage::trie {
     std::map<Buffer, std::optional<Buffer>, std::less<>> cache_;
     std::deque<Buffer> cleared_prefixes_;
     std::weak_ptr<TrieBatch> parent_;
+
+    friend struct TopperTrieCursor;
   };
 
+  /**
+   * Required for:
+   * - ext_storage_next_key_version_1
+   * - ext_default_child_storage_next_key_version_1
+   */
+  struct TopperTrieCursor : PolkadotTrieCursor {
+    TopperTrieCursor(std::shared_ptr<TopperTrieBatchImpl> batch,
+                     std::unique_ptr<PolkadotTrieCursor> cursor);
+
+    outcome::result<bool> seekFirst() override;
+    outcome::result<bool> seek(const BufferView &key) override;
+    outcome::result<bool> seekLast() override;
+    bool isValid() const override;
+    outcome::result<void> next() override;
+    outcome::result<void> prev() override;
+    std::optional<Buffer> key() const override;
+    std::optional<BufferOrView> value() const override;
+
+    outcome::result<void> seekLowerBound(const BufferView &key) override;
+    outcome::result<void> seekUpperBound(const BufferView &key) override;
+
+    void choose();
+    bool isRemoved() const;
+    outcome::result<void> skipRemoved();
+    outcome::result<void> step();
+
+    std::shared_ptr<TopperTrieBatchImpl> batch_;
+    std::unique_ptr<PolkadotTrieCursor> cursor_;
+    std::optional<Buffer> cursor_key_;
```

Comment:
Please comment that this is just cached cursor_->key(), because a separate variable for it raises suspicion that it might be not.